src/vos/tests/evt_ctl.c

```diff
@@ -1440,6 +1448,169 @@ test_evt_iter_delete_internal(void **state)
 	assert_int_equal(rc, 0);
 }
 
+static void
+test_evt_variable_record_size_internal(void **state)
+{
+	struct test_arg		*arg = *state;
+	daos_handle_t		 toh;
+	struct evt_entry_in	 entry = {0};
+	int			 rc, count;
+	int			 epoch;
+	const int		val[] = {D_1K_SIZE,
+					D_16K_SIZE};
+	uint64_t		data_size;
+	char                     *data;
+
+	rc = evt_create(ts_feats, 13, arg->ta_uma, arg->ta_root,
+		DAOS_HDL_INVAL, &toh);
+	assert_int_equal(rc, 0);
+	for (count = 0; count < sizeof(val)/sizeof(int); count++) {
+		/* Try to insert a bunch of entries with variable data sizes */
+		data_size = val[count];
+		D_ALLOC(data, data_size);
```

Comment:
FREE this allocated memory

Reply:
This memory is freed at the end of each loop. Added D_FREE for data and actual when the fail_msg is encountered.